File: Interview Preparation /6th Problem/Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
         if (prices == null || prices.length <= 1) {
            return 0; // No profit can be made with less than 2 prices
        }

        int minPrice = prices[0];
        int maxProfit = 0;

        for (int currentPrice : prices) {
            minPrice = Math.min(minPrice, currentPrice);
            int currentProfit = currentPrice - minPrice;
            maxProfit = Math.max(maxProfit, currentProfit);
        }

        return maxProfit;
    }
}

File: Practice Question/6th Problem/Best Time to Buy and Sell Stock.c
#define max(a,b) (a>b?a:b)
#define min(a,b) (a<b?a:b)
int maxProfit(int* prices, int pricesSize) {
    int minVal = INT_MAX;
    int result = 0;
    
    for(int i = 0 ; i < pricesSize ; i++ )
    {
        minVal = min(minVal, prices[i]);
        result = max(result, prices[i] - minVal);
    }
    return result;
}
